tests/parametric/test_otel_span_methods.py

```diff
@@ -396,6 +398,212 @@ def test_otel_set_attributes_separately(self, test_agent, test_library):
             assert span["name"] == "kafka.receive"
             assert span["resource"] == "operation"
 
+    @missing_feature(context.library < "java@1.26.0", reason="Implemented in 1.26.0")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library == "python_http", reason="Not implemented")
+    def test_otel_span_started_with_link_from_another_span(self, test_agent, test_library):
+        """Test adding a span link created from another span.
+        This tests the functionality of "create a direct link between two spans
+        given two valid span (or SpanContext) objects" as specified in the RFC.
+        """
+        with test_library:
+            with test_library.otel_start_span("root") as parent:
+                parent.end_span()
+                with test_library.otel_start_span(
+                    "child",
+                    parent_id=parent.span_id,
+                    links=[Link(parent_id=parent.span_id, attributes={"foo": "bar", "array": ["a", "b", "c"]})],
+                ) as child:
+                    child.end_span()
+
+        traces = test_agent.wait_for_num_traces(1)
+        trace = find_trace_by_root(traces, otel_span(name="root"))
+        assert len(trace) == 2
+
+        root = find_span(trace, otel_span(name="root"))
+        child = find_span(trace, otel_span(name="child"))
```

Comment:
nit: Spans are sorted by their start_time. We can just access the spans by index. The find_span method sometimes results in false matches. Honestly I think we can should remove it but that's beyond the scope of this PR.

```suggestion
        root = trace[0][0]
        child = trace[0][1]
```

Reply:
I find the `find_span()` method more explicit. Can we rather fix it? 
It is way more explicit for test source reader to understand to expected behavior rather than relying on an implicit order.

Reply:
Yup we should fix it. We can leave this test as is.